code-name-taran/src/com/spaghettisoft/component/ninemenmorris/BoardState.java

```diff
@@ -0,0 +1,344 @@
+            for (int i = 0; i < boardArrays.length; i++) {
+                for (int j = 0; j < boardArrays[i].length; j++) {
+
+                    if (boardArrays[i][j].getPoolType().equals(BoardTile.BLACK)) {
+                        poolCounterOponent++;
+                    }
+                }
+            }
+            // System.out.println("Black pools " + poolCounterOponent);
+            if ((poolCounterOponent < 3) && (placedPools == 18)) {
+                gameNotEnded = false;
+                System.out
+                        .println("White player wins !!! Black player has only " + poolCounterOponent + " pools left!");
+            }
+        } else {
+            System.out.println("EMPTY pools" + poolCounterOponent);
+            System.out.println("emty pull type");
+        }
+    }
+
+    @Override
+    protected void drawGame() {
+
+        for (int i = 0; i < boardArrays.length; i++) {
+            for (int j = 0; j < boardArrays[i].length; j++) {
+                System.out.print(boardArrays[i][j].getVisalization());
+            }
+            System.out.println();
+            System.out.println();
+        }
+        System.out.println("===================================");
+    }
+
+    @Override
+    protected void printEndGameMessage() {
+        drawGame();
+        initialize();
+    }
+
+    @Override
+    protected boolean isEnded() {
+        return !gameNotEnded;
+    }
+
+    @Override
+    protected void processGame() {
+        if (turnNumber < 19) {
+            processPlacePool(turnNumber);
+        } else {
+            processMovePool(turnNumber);
+        }
+        turnNumber++;
+    }
+
+    private void processMovePool(int j) {
+        if (j % 2 == 0) {
+            while (true) {
+                System.out.println("BLACK : Move FROM place: ");
+                String placeFrom = StaticObjects.scanner.nextLine().toUpperCase();
+
+                System.out.println("BLACK : Move TO place: ");
+                String placeTo = StaticObjects.scanner.nextLine().toUpperCase();
+
+                try {
+                    movePool(placeFrom, placeTo);
+                    break;
+
+                } catch (IllegalMoveException e) {
+                    System.out.println("Wrong move. Please try again.");
+                }
+            }
+
+        } else {
+
+            while (true) {
+                System.out.println("WHITE : Move FROM place: ");
+                String placeFrom = StaticObjects.scanner.nextLine().toUpperCase();
+
+                System.out.println("WHITE : Move TO place: ");
+                String placeTo = StaticObjects.scanner.nextLine().toUpperCase();
+                try {
+                    movePool(placeFrom, placeTo);
+                    break;
+                } catch (IllegalMoveException e) {
+                    System.out.println("Wrong move. Please try again.");
+                }
+            }
+        }
+    }

```

Comment:
There is a duplication of code here that can be removed by extracting the "WHITE" / "BLACK" string is separate variable like this: 
String playerColor = j % 2 == 0 ? "WHITE" : "BLACK";
....
System.out.println(playerColor + " : Move FROM place: ");
...
System.out.println(playerColor + " : Move TO place: ");